src/core/geometry/qgsgeos.h

```diff
@@ -547,10 +547,25 @@ class CORE_EXPORT QgsGeos: public QgsGeometryEngine
      * If \a edgesOnly is TRUE than line string boundary geometries will be returned
      * instead of polygons.
      * An empty geometry will be returned if the diagram could not be calculated.
+     * \see constrainedDelaunayTriangulation()
      * \since QGIS 3.0
      */
     QgsGeometry delaunayTriangulation( double tolerance = 0.0, bool edgesOnly = false, QString *errorMsg = nullptr ) const;
 
+    /**
+     * Returns a constrained Delaunay triangulation for the vertices of the geometry.
+     *
+     * An empty geometry will be returned if the triangulation could not be calculated.
+     *
+     * This method requires a QGIS build based on GEOS 3.11 or later.
+     *
+     * \throws QgsNotSupportedException on QGIS builds based on GEOS 3.10 or earlier.
+     * \see delaunayTriangulation()
+     *
+     * \since QGIS 3.36
```

Comment:
I want to get this in for release and backport -- it fixes a crash and silent data loss issue

Reply:
Mmmh ok, I hadn't read the ticket description, if it fixes crashes then OK for 3.36. Even though I've done it before, and it was allowed, I always prefer to avoid adding new methods in backports. Especially as there are several of them here, and not small ones. Anyway, they seem quite limited to this case, well tested, and, again, if it can improve QGIS behavior, that's fine with me. Nevertheless, would it be possible to do some manual testing before the merge of backport 3.34?

Reply:
Ultimately, I trust geos much more here vs poly2tri. You can see on that library's tracker that there's many known issues, and basically zero upstream maintainence happening. In contrast geos is very well supported, thoroughly tested, it's robust, and very actively maintained. If we find an issue I'm confident upstream will address promptly. 

I'd say given this the reward of backporting far outweighs the risk. (Especially because we already know if fixes at least two serious issues on the qgis side)

Reply:
Also worth pointing out that the complex looking new code for random points is identical to the existing poly2tri branch, just adapted to call geos for the tesselation and to adapt to the data structure returned by geos. The actual logic is unchanged.